src/main/java/org/jabref/logic/importer/fetcher/WorldcatFetcher.java

```diff
@@ -0,0 +1,212 @@
+        } catch (ParserConfigurationException e) {
+            throw new IllegalArgumentException ("Parser Config Exception: " + e.getMessage (), e);
+        } catch (SAXException e) {
+            throw new IllegalArgumentException ("SAX Exception: " + e.getMessage (), e);
+        } catch (IOException e) {
+            throw new IllegalArgumentException ("IO Exception: " + e.getMessage (), e);
+        }
+    }
+    
+    /**
+     * Parse OpenSearch XML to retrieve OCLC-ids and fetch details about those. 
+     * Returns new XML doc in the form
+     * <pre>
+     * {@literal <entries>}
+     *      {@literal <entry>}
+     *          OCLC DETAIL
+     *      {@literal </entry>}
+     *      ...
+     * {@literal </entries>}
+     * </pre>
+     * @param doc the open search result
+     * @return  the new xml document
+     * @throws FetcherException
+     */
+    private Document parseOpenSearchXML(Document doc) throws FetcherException {
+        try {
+            Element feed = (Element) doc.getElementsByTagName ("feed").item (0);
+            NodeList entryXMLList = feed.getElementsByTagName ("entry");
+
+            DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance ();
+            DocumentBuilder docBuilder = docFactory.newDocumentBuilder ();
+            
+            Document newDoc = docBuilder.newDocument ();
+
+            Element root = newDoc.createElement ("entries");
+            newDoc.appendChild (root);
+            for (int i = 0; i < entryXMLList.getLength (); i++) {
+                Element xmlEntry = (Element) entryXMLList.item (i);
+                
+                String oclc = xmlEntry.getElementsByTagName ("oclcterms:recordIdentifier").item (0).getTextContent ();
+                Element detailedInfo = getSpecificInfoOnOCLC (oclc);
+
+                Element newEntry = newDoc.createElement ("entry");
+                newEntry.appendChild (detailedInfo);
+
+                root.appendChild (newEntry);
+            }
+            return newDoc;
+        } catch (ParserConfigurationException e) {
+            throw new FetcherException("Error with XML creation (Worldcat fetcher)", e);
+        } catch (IOException e) {
+            throw new FetcherException("Error with OCLC parsing (Worldcat fetcher)", e);
+        }
+    }
+
+    @Override
+    public List<BibEntry> performSearch (BibEntry entry) throws FetcherException {
+        Optional<String> entryTitle = entry.getLatexFreeField (StandardField.TITLE);
+        if (entryTitle.isPresent ()) {
+            String openSearchXMLResponse = makeOpenSearchRequest (entryTitle.get ());
+            Document openSearchDocument = parse (openSearchXMLResponse);
+
+            Document detailedXML = parseOpenSearchXML (openSearchDocument);
+            String detailedXMLString;
+            try (StringWriter sw = new StringWriter ()){
+                //Transform XML to String
+                TransformerFactory tf = TransformerFactory.newInstance ();
```

Comment:
We have the convention that there shouldn't be a space between the method name and the brackets, so `newInstance()`. Could you please reformat the code accordingly. Thanks

Reply:
Sorry, then I misunderstood the style schema! Fixed